File: tags/releases/rl-glue-codec-c-1.0-alpha/tests/test_seeds_environment.c
/* 
* Copyright (C) 2007, Brian Tanner

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.

* 
*  $Revision$
*  $Date$
*  $Author$
*  $HeadURL$
* 
*/
#ifdef HAVE_CONFIG_H
#include <config.h>
#endif

#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <rlglue/Environment_common.h>
#include "useful_functions.h";


observation_t o={0};
reward_observation_t ro={0};

state_key_t env_saved_state_key={0};
random_seed_key_t env_saved_random_key={0};

task_specification_t env_init()
{    
	return "";
}

observation_t env_start()
{
	return o;
}

reward_observation_t env_step(action_t a)
{
  return ro;
}

void env_cleanup()
{

}

void env_set_state(state_key_t sk)
{
	copy_structure_to_structure(&env_saved_state_key, &sk);
}
     
void env_set_random_seed(random_seed_key_t rsk)
{
	copy_structure_to_structure(&env_saved_random_key, &rsk);
}

state_key_t env_get_state()
{
	return env_saved_state_key;
}

random_seed_key_t env_get_random_seed()
{
	return env_saved_random_key;
}

message_t env_message(const message_t inMessage) {

}
	

File: projects/codecs/Python/src/tests/test_seeds_environment.py
# 
# Copyright (C) 2007, Mark Lee
# 
#http://rl-glue-ext.googlecode.com/
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

import random
import sys
from rlglue.environment.Environment import Environment
from rlglue.environment import EnvironmentLoader as EnvironmentLoader
from rlglue.types import Observation
from rlglue.types import Action
from rlglue.types import Reward_observation
from rlglue.types import State_key
from rlglue.types import Random_seed_key

class test_seeds_environment(Environment):
	savedStateSeed=State_key()
	savedRandomSeed=Random_seed_key()
	
	o=Observation()
	
	def env_init(self):  
		return ""
	
	def env_start(self):
		return self.o;
	
	def env_step(self,action):
		terminal=False
		ro=Reward_observation()
		ro.r=1.0
		ro.o=self.o
		ro.terminal=terminal
		return ro	

	def env_cleanup(self):
		pass
	
	def env_set_state(self, stateKey):
		self.savedStateKey=stateKey
		
	def env_set_random_seed(self, randomSeedKey):
		self.savedRandomSeed=randomSeedKey
	
	def env_get_state(self):
		return self.savedStateKey
	
	def env_get_random_seed(self):
		return self.savedRandomSeed
	
	def env_message(self,inMessage):
		return None

if __name__=="__main__":
	EnvironmentLoader.loadEnvironment(test_seeds_environment())